packages/linter/src/executors/eslint/utility/eslint-utils.ts

```diff
@@ -1,50 +1,90 @@
-import { ESLint } from 'eslint';
+import type { ESLint } from 'eslint';
 import type { Schema } from '../schema';
 
-export async function loadESLint() {
-  let eslint;
+async function resolveESLintClass(
+  useFlatConfig = false
+): Promise<typeof ESLint> {
   try {
-    eslint = await import('eslint');
-    return eslint;
+    if (!useFlatConfig) {
+      return (await import('eslint')).ESLint;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { FlatESLint } = require('eslint/use-at-your-own-risk');
+    return FlatESLint;
   } catch {
     throw new Error('Unable to find ESLint. Ensure ESLint is installed.');
   }
 }
 
-export async function lint(
+export async function resolveAndInstantiateESLint(
   eslintConfigPath: string | undefined,
-  options: Schema
-): Promise<ESLint.LintResult[]> {
-  const projectESLint: { ESLint: typeof ESLint } = await loadESLint();
+  options: Schema,
+  useFlatConfig = false
+) {
+  if (
+    useFlatConfig &&
+    eslintConfigPath &&
+    !eslintConfigPath?.endsWith('eslint.config.js')
```

Comment:
Nit: The `optional chaining` on like 27 is not needed, since we check for `eslintConfigPath`'s existence on line before